packages/eventual-send/src/async-contexts/sync-context-weak-transpose.js

```diff
@@ -0,0 +1,40 @@
+// Based on sync-context-weak-inline.js
+
+// The transposed representation places most of the mutable state in
+// the `SyncContext` instances. While `__keys__` remains a top-level mutable
+// variable, the possible value of `__keys__` at any time is only an
+// immutable linked list of immutable empty `key` objects.
+
+// eslint-disable-next-line no-underscore-dangle
+let __keys__;
+
+export class SyncContext {
+  #transposedMap = new WeakMap();
+
+  run(val, cb, args = []) {
+    const key = harden({});
+    this.#transposedMap.set(key, val);
+    const next = harden({ key, prevKeys: __keys__ });
+
+    const prev = __keys__;
+    try {
+      __keys__ = next;
+      return cb(...args);
+    } finally {
+      __keys__ = prev;
+    }
+  }
+
+  get() {
+    let keys = __keys__;
+    while (keys !== undefined) {
```

Comment:
this could be open to a timing attack for communicating across 2 parties Eve and Mallory

Assuming Eve has access to a timer, she can use setTimeout to measure event loop delay caused by a `get()` call.
Meanwhile, when a condition Mallory wants to communicate to Eve is met, Mallory uses `run()` significantly many times to make a noticeable delay.